infra/nightly-resources/index.js

```diff
@@ -1,46 +1,194 @@
-// Top-level load function for the main index page.
-function load() {
-  // data should be in the format of Array<{runMethod: String, benchmark: String, total_dyn_inst: Int, hyperfine: Array<{results: **hyperfine `--json` results**}>}>
-  console.log(data);
-
-  let container = document.getElementById("profile");
-  let parsed = [];
-
-  // aggregate benchmark runs into a list by their "benchmark" key
-  let groupedBy = {};
-  data.forEach((obj) => {
-    if(!groupedBy[obj.benchmark]) {
-      groupedBy[obj.benchmark] = [];
+async function getPreviousRuns() {
+    const req = await fetch("https://nightly.cs.washington.edu/reports-json/eggcc/");
+    const files = await req.json();
+    
+    // map allLinks into an objects of the shape:
+    // {branch: <git branch:string>, commit: <git commit:string>, date: <unix timestamp:int>, url: <absolute url to nightly page:string>}
+    const comparisons = [];
+    for (let i = 1; i < files.length; i++) {
+        // file name is of the format <date>:"nightly":<branch>:<commit>
+        const [date, _, branch, commit] = files[i].name.split(":");
+
+        const run = {
+            branch: branch,
+            commit: commit,
+            // type coerce a unix timestamp that is a string into an int with a `+`
+            date: +date,
+            // The file server only hands us back the directory names,
+            // but we want to make sure that we only use absolute URLs so that we can run this page
+            // in any environment (local or otherwise)
+            url: `https://nightly.cs.washington.edu/reports/eggcc/${files[i].name}`,
+        }
+        
+        comparisons.push(run);
+    }
+
+    comparisons.sort((l, r) => {
+        if (l.date < r.date) {return -1; }
+        if (l.date > r.date) {return 1; }
+        return 0;
+    });
+
+    return comparisons;
+}
+
+async function buildNightlyDropdown(element, previousRuns, initialIdx) {
+    const select = document.getElementById(element);
+    
+    const formatRun = (run) => `${run.branch} - ${run.commit}`
```

Comment:
including the date would be quite helpful

Reply:
:+1: